base/iobuffer.jl

```diff
@@ -260,22 +260,29 @@ bytesavailable(io::GenericIOBuffer) = io.size - io.ptr + 1
 position(io::GenericIOBuffer) = io.ptr - io.offset - 1
 
 function skip(io::GenericIOBuffer, n::Integer)
-    seekto = io.ptr + n
-    n < 0 && return seek(io, seekto-1) # Does error checking
-    io.ptr = min(seekto, io.size+1)
-    return io
+    skip(io, clamp(n, Int))
+end
+function skip(io::GenericIOBuffer, n::Int)
+    if signbit(n)
+        seekto = clamp(widen(position(io)) + widen(n), Int)
+        seek(io, seekto) # Does error checking
+    else
+        @assert io.ptr ≤ io.size + 1
```

Comment:
Don't use `@assert` for hard requirements of correctness, as it's valid to be removed.

Reply:
It's ok for this assert to be removed. I can't see a case where it does anything.